### ISSUE TITLE
[0.66] React-Native-Popup-Menu not closing anymore

### ISSUE THREAD

**Issue by creambyemute** [microsoft/react-native-windows#8966]
### Problem Description

We are using https://github.com/instea/react-native-popup-menu (JS-only module) to render ContextMenus ActionMenus or DropDowns.

With RNW 0.66 the menu does open, but is not closable anymore. Interaction with other elements than the Menus children does not work. Backdrop (onDismiss on click outside) does not work anymore either it seems.

I have tested with 0.63, 0.64 and 0.65 and in these RN + RNW versions closing the menu works.

### Steps To Reproduce

1. `npx react-native init test066RNWPopupMenu`
2. `cd  test066RNWPopupMenu`
3. `npx react-native-windows-init --overwrite`
4. `yarn add react-native-popup-menu`
5. See code in  `Snack, code example, screenshot, or link to a repository` for an reproduction sample

Doing the same for 0.63/0.64/0.65, there it works.
For Android/iOS it works regardless of the RN Version.

### Expected Results

Menu is openable as well as closable.

### CLI version

6.1.0

### Environment

```markdown
System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
    Memory: 22.16 GB / 31.86 GB
  Binaries:
    Node: 14.17.6 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.15 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK:
      API Levels: 26, 27, 28, 29, 30
      Build Tools: 28.0.0, 28.0.1, 28.0.2, 28.0.3, 29.0.0, 29.0.1, 29.0.2, 29.0.3, 30.0.0, 30.0.1, 30.0.2, 30.0.3
      System Images: android-26 | Google APIs Intel x86 Atom, android-26 | Google APIs Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom, android-27 | Google Play Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom, android-28 | Google Play Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-29 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
      Android NDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.10586.0, 10.0.14393.0, 10.0.17763.0, 10.0.18362.0, 10.0.19041.0
  IDEs:
    Android Studio: Version  4.0.0.0 AI-193.6911.18.40.6514223
    Visual Studio: 16.8.30804.86 (Visual Studio Community 2019)
  Languages:
    Java: 13.0.2 - C:\Program Files\OpenJDK\jdk-13.0.2\bin\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2
    react-native: 0.66.1 => 0.66.1
    react-native-windows: 0.66.2 => 0.66.2
  npmGlobalPackages:
    *react-native*: Not Found
```

### Target Platform Version

10.0.16299

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2019

### Build Configuration

Release

### Snack, code example, screenshot, or link to a repository

Init App.js with rn-popup-menu added, stripped away section stuff.

```javascript
import React from 'react';
import type {Node} from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  useColorScheme,
  View,
  Button,
  GestureResponderEvent,
} from 'react-native';

import {
  Colors,
  Header,
  LearnMoreLinks,
} from 'react-native/Libraries/NewAppScreen';

import {
  MenuProvider,
  Menu,
  MenuTrigger,
  MenuOptions,
  MenuOption,
  renderers,
} from 'react-native-popup-menu';
const {NotAnimatedContextMenu} = renderers;

const menuProviderCustomStyles = {
  backdrop: {backgroundColor: 'red'},
};

const App: () => Node = () => {
  const menuRef = React.useRef(undefined);
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  const onShowMenu = () => {
    menuRef?.current?.open();
  };

  const onHideMenu = () => {
    menuRef?.current?.close();
  };

  const onPressShowMenuButton = (event: GestureResponderEvent) => {
    onShowMenu();
  };

  return (
    <MenuProvider backHandler={true} customStyles={menuProviderCustomStyles}>
      <SafeAreaView style={backgroundStyle}>
        <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
        <ScrollView
          contentInsetAdjustmentBehavior="automatic"
          style={backgroundStyle}>
          <Header />
          <View
            style={{
              backgroundColor: isDarkMode ? Colors.black : Colors.white,
            }}>
            <Button title={'Show Menu'} onPress={onPressShowMenuButton} />
            <Menu ref={menuRef} renderer={NotAnimatedContextMenu}>
              <MenuTrigger />
              <MenuOptions>
                <MenuOption onSelect={onHideMenu} text="Save" />
                <MenuOption onSelect={onHideMenu} text="Delete" />
              </MenuOptions>
            </Menu>
            <LearnMoreLinks />
          </View>
        </ScrollView>
      </SafeAreaView>
    </MenuProvider>
  );
};

export default App;
```

**Comment by jonthysell** [comment]
@TatianaKapos This looks like something that may be happening due to missing mouse/pointer events.

Can you see if this still repos in 0.69?

**Comment by TatianaKapos** [comment]
@jonthysell
Seems to be fixed in 0.69 playground, menu closes via clicking on Menu, Item, and Backdrop

https://user-images.githubusercontent.com/42554868/167962398-6d3e8aaa-87b3-4f34-b452-3da6a694ceb8.mp4
